### PR TITLE
fix: Crash when bluetooth is off and Send button is clicked

### LINKED ISSUE #38: Crash when bluetooth is off and Send button is clicked
**Describe the bug**
Bluetooth adapter is not turned on error is showing up as there's no check in the code.

**To Reproduce**
Turn off bluetooth and click send button.

**Expected behavior**
App should not crash. 

**Logs**
AndroidRuntime: FATAL EXCEPTION: main
    Process: com.nilhcem.blenamebadge, PID: 15932
    java.lang.IllegalStateException: BT Adapter is not turned ON
        at no.nordicsemi.android.support.v18.scanner.BluetoothLeUtils.checkAdapterStateOn(BluetoothLeUtils.java:136)
        at no.nordicsemi.android.support.v18.scanner.BluetoothLeScannerImplLollipop.startScanInternal(BluetoothLeScannerImplLollipop.java:57)
        at no.nordicsemi.android.support.v18.scanner.BluetoothLeScannerCompat.startScan(BluetoothLeScannerCompat.java:112)
        at com.nilhcem.blenamebadge.device.bluetooth.ScanHelper.startLeScan(ScanHelper.kt:57)
        at com.nilhcem.blenamebadge.ui.message.MessagePresenter.sendBytes(MessagePresenter.kt:36)
        at com.nilhcem.blenamebadge.ui.message.MessagePresenter.sendBitmap(MessagePresenter.kt:25)
        at com.nilhcem.blenamebadge.ui.message.MessageActivity$onCreate$3.onClick(MessageActivity.kt:69)
        at android.view.View.performClick(View.java:6291)
        at android.view.View$PerformClick.run(View.java:24931)
        at android.os.Handler.handleCallback(Handler.java:808)
        at android.os.Handler.dispatchMessage(Handler.java:101)
        at android.os.Looper.loop(Looper.java:166)
        at android.app.ActivityThread.main(ActivityThread.java:7529)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)

**Smartphone Info:**

|Device              | Huawei Honor 7x |
|Android Version| Oreo 8.1    |

**Additional context**
Need to check if the bluetooth is enabled before starting the scan.

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by balaji-ramavathu**
Fixes #38 

Changes: Added a check before scan() is called to check if the bluetooth is enabled or not.

### CODE REVIEW

**Comment by balaji-ramavathu** [comment]
@mariobehling please review :-)

File: app/src/main/java/com/nilhcem/blenamebadge/device/bluetooth/ScanHelper.kt [hunk 1]
```diff
@@ -54,7 +55,12 @@ class ScanHelper {
                 .setScanMode(ScanSettings.SCAN_MODE_LOW_LATENCY)
                 .build()
 
-        scanner.startScan(filters, settings, scanCallback)
+        //check if bluetooth is enabled
+        if (BluetoothAdapter.getDefaultAdapter().isEnabled) {
+            scanner.startScan(filters, settings, scanCallback)
+        } else {
+            Timber.i { "Turn on Bluetooth and try again" }
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/device/bluetooth/ScanHelper.kt]: There's no information to user that what he/she should do!

Show a toast

Comment (balaji-ramavathu) [reply app/src/main/java/com/nilhcem/blenamebadge/device/bluetooth/ScanHelper.kt]: Done!

File: app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt [hunk 2]
```diff
@@ -56,19 +56,23 @@ class MessageActivity : AppCompatActivity() {
...
-                    runOnUiThread { send.isClickable = true }
-                }
-            }, SCAN_TIMEOUT_MS)
-            if (content.text.isEmpty()) {
-                presenter.sendBitmap(this, BitmapFactory.decodeResource(resources, R.drawable.mix2))
-            } else {
-                presenter.sendMessage(this, convertToDeviceDataModel())
-            }
+           if (BluetoothAdapter.getDefaultAdapter().isEnabled) {
+               // Easter egg
```
Comment (yashk2000) [inline app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt]: @balaji-ramavathu please correct the indentation here.

**Comment by balaji-ramavathu** [comment]
I've messed up this PR so much, so I'm closing it and open a new one.

**Pull request closed without merging**